front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -14,10 +14,75 @@ import type {
   SimulationResponseSuccess,
 } from 'applications/operationalStudies/types';
 import type { ReportTrainV2 } from 'common/api/osrdEditoastApi';
-import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+import { mmToKm, mmToM, msToKmh, mToKm } from 'utils/physics';
 
 import { electricalProfilesDesignValues } from './consts';
 
+type SpeedLimitTagInput = {
+  boundaries: number[];
+  values: {
+    speed: number;
+    source: { speed_limit_source_type: string; tag?: string } | null;
+  }[];
+};
+
+const getTagName = (source: { speed_limit_source_type: string; tag?: string } | null): string => {
+  if (source === null) {
+    return 'missing_from_train';
+  }
+  switch (source.speed_limit_source_type) {
+    case 'given_train_tag':
+      return source.tag || '';
+    case 'fallback_tag':
+      return source.tag || '';
+    case 'unknown_tag':
+      return 'incompatible';
+    case 'UU':
+      return source.tag || '';
+    default:
+      return 'missing_from_train'; // Fallback for any unknown types
+  }
+};
+
+const getTagColor = (sourceType: string): string => {
+  switch (sourceType) {
+    case 'given_train_tag':
+      return '#494641'; // Color for given train tags
+    case 'fallback_tag':
+      return '#216482'; // Color for fallback tags
+    case 'unknown_tag':
+      return '#EAA72B'; // Red color for incompatible tags (alerts)
+    case 'UU':
+      return '#D91C1C';
+    default:
+      return '#94918E'; // Default color for missing or unknown tags
+  }
+};
+
+export const formatSpeedLimitTags = (
+  input: SpeedLimitTagInput,
+  pathLength: number
+): LayerData<SpeedLimitTagValues>[] => {
+  const positions: LayerData<SpeedLimitTagValues>[] = [];
+
+  for (let i = 0; i < input.values.length; i += 1) {
+    const start = i === 0 ? 0 : input.boundaries[i - 1];
+    const end = i === input.values.length - 1 ? pathLength : input.boundaries[i];
+
+    positions.push({
+      position: {
+        start: mToKm(start),
+        end: mToKm(end),
+      },
+      value: {
+        tag: getTagName(input.values[i].source), // This will be 'incompatible' for unknown_tag
+        color: getTagColor(input.values[i].source?.speed_limit_source_type || ''), // This should set the correct color
```

Comment:
I think you can merge the two functions getTagName and getTagColor => getTag which returns the name and the color, what do you think ?